Tools/scripts/build_ci.sh

```diff
@@ -424,6 +424,13 @@ for t in $CI_BUILD_TARGET; do
         continue
     fi
 
+    if [ "$t" == "pre-commit-cleanliness" ]; then
+        echo "Checking pre-commit code cleanliness"
+        pre-commit install
```

Comment:
@Ryanf55 that means that after a first run precommit will be enable on git commit. Probably not the best move. Just doing the pre-commit run is enough and don't mess the git config for now (I still want pre-commit)

Reply:
Are you saying that local developers run build_ci with specific arguments to hit this code block?

Reply:
@Ryanf55 I think Pierre is saying we shouldn't be making this permanent change to their setup in here.

We can do it as part of the machine setup in the github workflows instead, yes?

Reply:
yes, but I thought it would be ok because this is a CI script. I'm happy to move it if you think we need to worry about devs running this check and then being surprised pre-commit is now enabled.

Reply:
Yes, please move it.

Reply:
Done! Let's see how it goes...

Reply:
@Ryanf55 normally there is not issue to use `pre-commit run --all-files` on CI, install isn't need for CI